cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -117,219 +96,42 @@ func (m *Manager) Destroy(ctx context.Context, options DestroyOptions) (*Destroy
-	}
-
-	prompters := Prompters{
-		Location:                   m.promptLocation,
-		Subscription:               m.promptSubscription,
-		EnsureSubscriptionLocation: m.ensureSubscriptionLocation,
-	}
-
-	infraProvider, err := NewProvider(
-		ctx,
-		console,
-		azCli,
-		commandRunner,
-		env,
-		projectPath,
-		infraOptions,
-		prompters,
-		principalProvider,
-		alphaFeatureManager,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("error creating infra provider: %w", err)
-	}
-
-	requiredTools := infraProvider.RequiredExternalTools()
-	if err := tools.EnsureInstalled(ctx, requiredTools...); err != nil {
-		return nil, err
-	}
-
-	m.provider = infraProvider
-	if err := m.provider.EnsureConfigured(ctx); err != nil {
-		return nil, err
-	}
-
-	return m, nil
 }
 
-func (m *Manager) promptSubscription(ctx context.Context, msg string) (subscriptionId string, err error) {
-	return promptSubscription(ctx, msg, m.console, m.accountManager)
-}
-
-func (m *Manager) promptLocation(
-	ctx context.Context,
-	subId string,
-	msg string,
-	filter func(loc account.Location) bool,
-) (string, error) {
-	return promptLocation(ctx, subId, msg, filter, m.console, m.accountManager)
-}
-
-func (m *Manager) ensureSubscriptionLocation(ctx context.Context, env *environment.Environment) error {
-	return EnsureEnv(ctx, m.console, m.env, m.accountManager)
-}
-
-type CurrentPrincipalIdProvider interface {
-	// CurrentPrincipalId returns the object id of the current logged in principal, or an error if it can not be
-	// determined.
-	CurrentPrincipalId(ctx context.Context) (string, error)
-}
+func (m *Manager) newProvider(ctx context.Context) (Provider, error) {
+	if m.options.Provider == "" {
+		m.options.Provider = Bicep
+	}
 
-type principalIDProvider struct {
-	env                *environment.Environment
-	userProfileService *azcli.UserProfileService
-	subResolver        account.SubscriptionTenantResolver
-}
+	if alphaFeatureId, isAlphaFeature := alpha.IsFeatureKey(string(m.options.Provider)); isAlphaFeature {
+		if !m.alphaFeatureManager.IsEnabled(alphaFeatureId) {
+			return nil, fmt.Errorf("provider '%s' is alpha feature and it is not enabled. Run `%s` to enable it.",
+				m.options.Provider,
+				alpha.GetEnableCommand(alphaFeatureId),
+			)
+		}
 
-func (p *principalIDProvider) CurrentPrincipalId(ctx context.Context) (string, error) {
-	tenantId, err := p.subResolver.LookupTenant(ctx, p.env.GetSubscriptionId())
-	if err != nil {
-		return "", fmt.Errorf("getting tenant id for subscription %s. Error: %w", p.env.GetSubscriptionId(), err)
+		m.console.WarnForFeature(ctx, alphaFeatureId)
 	}
 
-	principalId, err := azureutil.GetCurrentPrincipalId(ctx, p.userProfileService, tenantId)
+	var provider Provider
+	err := m.serviceLocator.ResolveNamed(string(m.options.Provider), &provider)
 	if err != nil {
-		return "", fmt.Errorf("fetching current user information: %w", err)
+		return nil, fmt.Errorf("resolving provider '%s': %w", m.options.Provider, err)
```

Comment:
Suspect the most common (only?) error we are going to see here is when m.options.Provider is a string for a provider we don't support.  Just wondering if there's a better error experience for that case to give a message back to the user around what they need to do to fix things (in this case, I assume it's edit azure.yaml and fix something in `infra` section).